ogame/__init__.py

```diff
@@ -1154,64 +1154,49 @@ def spyreports(self, firstpage=1, lastpage=30):
             api_code = bs4.find('span', 'icon_apikey')['title']
             api_code = re.search(r'value=\'(.+?)\'', api_code).group(1)
             resources_data = {}
-            for li in ul.find_all('li'):
-                resource_name = li.find('div')['class']
+            for resource in resources_list.find_all('li'):
+                resource_name = resource.find('div')['class']
                 resource_name.remove('resourceIcon')
-                resources_data[resource_name[0]] = int(li['title'].replace('.', ''))
-            spied_fleet = {}
-            ul = bs4.find('ul', {'data-type': 'ships'})
-            for li in ul.find_all('li'):
-                tech_id = int(re.search(r'([0-9]+)', li.find('img')['class'][0]).group(1))
-                tech_amount = int(li.find('span', 'fright').text.replace('.', ''))
-                if tech_id in [212, 217]:
-                    tech_name = const.buildings.building_name(
-                        (tech_id, None, None)
-                    )
-                else:
-                    tech_name = const.ships.ship_name(
-                        (tech_id, None, 'shipyard')
-                    )
-                spied_fleet.update({tech_name: tech_amount})
-            spied_defences = {}
-            ul = bs4.find('ul', {'data-type': 'defense'})
-            for li in ul.find_all('li'):
-                tech_id = int(re.search(r'([0-9]+)', li.find('img')['class'][0]).group(1))
-                tech_name = const.buildings.defense_name(
-                    (tech_id, None, 'defenses')
-                )
-                tech_amount = int(li.find('span', 'fright').text.replace('.', ''))
-                spied_defences.update({tech_name: tech_amount})
-            spied_buildings = {}
-            ul = bs4.find('ul', {'data-type': 'buildings'})
-            for li in ul.find_all('li'):
-                tech_id = int(re.search(r'([0-9]+)', li.find('img')['class'][0]).group(1))
-                tech_name = const.buildings.building_name(
-                    (tech_id, None, None)
-                )
-                tech_level = int(li.find('span', 'fright').text.replace('.', ''))
-                spied_buildings.update({tech_name: tech_level})
-            spied_research = {}
-            ul = bs4.find('ul', {'data-type': 'research'})
-            for li in ul.find_all('li'):
-                tech_id = int(re.search(r'([0-9]+)', li.find('img')['class'][0]).group(1))
-                tech_name = const.research.research_name(
-                    (tech_id, None, 'research')
-                )
-                tech_amount = int(li.find('span', 'fright').text.replace('.', ''))
-                spied_research.update({tech_name: tech_amount})
+                resources_data[resource_name[0]] = int(resource['title'].replace('.', ''))
+
+            def get_tech_qt(tech_type):
```

Comment:
what does qt stand for?
needs bedder name

Reply:
It stands for quantity. I'm gonna rename that